integrationtests/src/test/java/org/axonframework/integrationtests/queryhandling/QueryThreadingIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+
+
+        QueryBus localQueryBus = SimpleQueryBus.builder().build();
+        queryBus = AxonServerQueryBus.builder()
+                .axonServerConnectionManager(connectionManager)
+                .configuration(configuration)
+                .localSegment(localQueryBus)
+                .updateEmitter(localQueryBus.queryUpdateEmitter())
+                .messageSerializer(serializer)
+                .genericSerializer(serializer)
+                .build();
+        queryBus.start();
+
+        queryBus.subscribe("query-b", String.class, query -> {
+            while (secondaryQueryBlock.get()) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+            return "b";
+        });
+
+        queryBus.subscribe("query-a", String.class, query -> {
+            QueryResponseMessage<String> b = queryBus.query(new GenericQueryMessage<>("start", "query-b", ResponseTypes.instanceOf(String.class))).get();
+            return "a" + b.getPayload();
+        });
+    }
+
+    @AfterEach
+    void tearDown() {
+        commandBus.shutdownDispatching();
+        queryBus.shutdownDispatching();
+
+        commandBus.disconnect();
+        queryBus.disconnect();
+
+        connectionManager.shutdown();
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void canStillHandleQueryResponsesWhileManyQueriesHandling() throws InterruptedException {
+
+        CompletableFuture<QueryResponseMessage<String>> query1 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query2 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query3 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query4 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query5 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query6 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+
+        Thread.sleep(500);
+        // We should still have the queries not done, it's waiting on the secondary one.
+        assertFalse(query1.isDone());
+        assertFalse(query2.isDone());
+        assertFalse(query3.isDone());
+        assertFalse(query4.isDone());
+        assertFalse(query5.isDone());
+        assertFalse(query6.isDone());
+
+        secondaryQueryBlock.set(false);
+
+        await().untilAsserted(() -> {
```

Comment:
This would mean it blocks indefinitely, right? Would make sense to have a max time window for this assertion.

Reply:
Awaitility has a default timeout of 10 seconds :) 
If you'd like I can add it anyway

Reply:
Added it at 5 seconds.